InputGroup component: Incorrect type definition for children prop

### Description

# When I
- Use the input-group.tsx snippet in a React component
(Please note that all npm module versions were taken to latest version to ensure the issue is specific to the input-group.tsx snippet and not caused by outdated dependencies.)
# I expected
- The project should build
# but
- It doesn't build, and complains:
```shell
❯ npm run build

> vite-react-typescript-starter@0.0.0 build
> tsc -b && vite build

src/components/ui/input-group.tsx:29:66 - error TS2345: Argument of type 'ReactElement<unknown, string | JSXElementConstructor<any>>' is not assignable to parameter of type 'ReactElement<InputElementProps, string | JSXElementConstructor<any>>'.
  Type 'unknown' is not assignable to type 'InputElementProps'.
    Type 'unknown' is not assignable to type 'Omit<PatchHtmlProps<Omit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "ref">>, "color" | ... 808 more ... | keyof PolymorphicProps>'.

29       React.Children.only<React.ReactElement<InputElementProps>>(children);
                                                                    ~~~~~~~~

src/components/ui/input-group.tsx:43:11 - error TS2698: Spread types may only be created from object types.

43           ...children.props,
             ~~~~~~~~~~~~~~~~~


Found 2 errors.


~/projects/chakra-ui-v3 2s
❯ 
```


### Link to Reproduction

https://stackblitz.com/edit/chakra-ui-v3-k6jtupjv?file=src%2Fcomponents%2Fui%2Finput-group.tsx

### Steps to reproduce

1. Go to the terminal
2. build the project
```shell
> npm run build
```
3. See error


### Chakra UI Version

3.2.4

### Browser

_No response_

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

I've observed that modifying the children prop type in the input-group.tsx snippet from `React.ReactElement` to `React.ReactElement<InputElementProps>` resolves the type errors.

I'm willing to submit a Pull Request to address this issue by:

- Updating the children prop type in input-group.tsx to React.ReactElement<InputElementProps>.
- Adding a basic test case to ensure type safety and prevent future regressions.

I would appreciate feedback on this approach and any guidance on the best way to contribute to the Chakra UI project, and specially on the later point, on where to add that basic test in a place were similar (i.e. typescript related) things are tested.

--- a/apps/compositions/src/ui/input-group.tsx
+++ b/apps/compositions/src/ui/input-group.tsx
@@ -7,7 +7,7 @@ export interface InputGroupProps extends BoxProps {
   endElementProps?: InputElementProps
   startElement?: React.ReactNode
   endElement?: React.ReactNode
-  children: React.ReactElement
+  children: React.ReactElement<InputElementProps>
   startOffset?: InputElementProps["paddingStart"]
   endOffset?: InputElementProps["paddingEnd"]
 }

